### ISSUE TITLE
Bug: [consistent-type-imports] "fixStyle"-option not honored

### ISSUE THREAD

**Issue by paleite** [typescript-eslint/typescript-eslint#7445]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.1.6&fileType=.ts&code=JYWwDg9gTgLgBDAnmApnA3nAYhCcC%2BcAZlBCHAOQ4QUDcAUKJLBgsmtQcaeVbnfSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyoMAbUiJo0KZAA0mcNhwAzSgA8AykSToozRk0RCS18ZOmotEAL4BdLS5AugA&tsconfig=&tokens=false

### Repro Code

```typescript
import type { Foo } from 'Foo';
import { type Foo } from 'Foo';
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/consistent-type-imports": ["error", {
      "fixStyle": "inline-type-imports"
    }]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expected the 1st line to produce an error saying that type should be inlined.

### Actual Result

No error at all.

### Additional Info

The same result when configuring the opposite (i.e. changing from "inline-type-imports" to "separate-type-imports").

**Comment by bradzacher** [comment]
It seems you've misunderstood the option.
The the rule doesn't concern itself with enforcing *how* you mark an import as type-only - just that you do.
The `fixStyle` option is there to control what the rule auto-fixes to only.

If you want to enforce inline vs top-level imports there is a rule mentioned [in the docs](https://typescript-eslint.io/rules/consistent-type-imports#related-to) - specifically the [`import/consistent-type-specifier-style`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md) rule
